processor/README.md

```diff
@@ -71,3 +81,50 @@ processors:
 ```
 Refer to [config.yaml](attributes/testdata/config.yaml) for detailed
 examples on using the processor.
+
+## <a name="fan-out"></a>Fan Out Processor
+<FILL ME IN - I'M LONELY!>
+
+## <a name="multi-consumer"></a>Multi-Consumer Processor
+<FILL ME IN - I'M LONELY!>
+
+## <a name="node-batcher"></a>Node Batcher Processor
+<FILL ME IN - I'M LONELY!>
+
+## <a name="probabilistic-sampler"></a>Probabilistic Sampler Processor
+<FILL ME IN - I'M LONELY!>
+
+## <a name="queued"></a>Queued Processor
+<FILL ME IN - I'M LONELY!>
+
+## <a name="span"></a>Span Processor
+The span processor modifies top level settings of a span. Currently, only 
+renaming a span is supported. 
+
+### Rename a span
+It takes a list of `keys` and an optional `separator` string. The attribute
+value for the keys are used to create a new name in the order specified in the
+configuration. If a separator is specified, it will separate values.
+
+For more information, refer to [config.go](span/config.go)
+```yaml
+span:
+  rename:
```

Comment:
The "rename" operation that we have now is pretty specialized. Would it make more sense to have a "span-rename" processor that does this specialized thing only? I think the answer also depends on what other operations we may add in the future to "span" processor. If you have any ideas it would be good to discuss them before committing to this particular config structure.

Reply:
In terms of other span operations, there aren't any that I can think of at the moment.  @flands  any ideas? 

That being said, this does set the precedence to add span modifications to this processor only instead of creating several processors that operate on different parts of the span. Kind of like Add Attributes/ Attribute Key they ended up having similar logic but it wasn't addressed because the precedent was set to add a processor per modification of an attribute.

One thing that I'd like to propose adding in a future pr is the concept of exclude/include where spans are either excluded/included from the processor based on service name and attribute values. Keeping rename as a field of span keeps the config clear. 
In more concrete terms it would look like:
```
span:
  exclude:
    service_names: [svc.a, svc.b]
    attributes:
      - key: instrumentation_version
         value: 123
      - key:  bob
         value: foo
  rename:
   <....the rest of the config>
```

Reply:
exclude/include idea sounds good to me. It will be uniform in the config and implemented once in the code, good all around.

I think we did a good thing that we combined all attribute modifications into one processor. I have an idea about changing in-memory representation of spans and this decision to have one processor will likely enable the change and allow to improve performance of the service.

Not sure about `span` processor. Let's solicit some opinions and possible ideas about what can be done with a span that is not a `rename` and is not an attribute modification.

Reply:
We did the `attributes` based on having a single processor to make changes to attributes, it seems reasonable to me to do similar for both attribute and span processors.

Reply:
@pjanotti I agree if we assume that will be many different operations that we can do on a span. Can you think of an example operation different from `rename`?

Reply:
@tigrannajaryan I can see a processor changing span Status or Kind based on some other specific property of the span. The other more complex structs like annotations, likely deserve a dedicated processor (similar to what we did for attributes).

Reply:
I've changed the field to be `name` for the configuration.